packages/react-dialog/Spec.md

```diff
@@ -2,62 +2,303 @@
+| open     | boolean                       | `false`   | Set to `true` when the dialog is visible |
+| overlay  | _slot_                        | undefined | Dimmed background of dialog              |
+
+> ⚠️ _Pending issue: `draggable`, the dragging functionality of the dialog is under consideration. This might be excluded from the dialog implementation._
+
+- `type` property (dialog variations):
+
+  - `modal`: When this type of dialog is open, the rest of the page is dimmed out and cannot be interacted with. The tab sequence is kept within the dialog and moving the focus outside the dialog will imply closing it. This is the default type of the component.
+
+  - `non-modal`: When a non-modal dialog is open, the rest of the page is not dimmed out and users can interact with the rest of the page. This also implies that the tab focus can move outside the dialog when it reaches the last focusable element.
+
+  - `alert`: is a special type of modal dialogs that interrupts the user's workflow to communicate an important message or ask for a decision. Unlike a typical modal dialog, the user must take an action though the options given to dismiss the dialog, and it cannot be dismissed through the dimmed background or esacpe key.
+
+### DOM
+
+```html
+<div role="dialog" class="fui-dialog">{children}</div>
+```
+
+## Dialog Header
+
+The DialogHeader component will expect to have a dialog title/header and will show by default the close (X icon) button.
+
+### DOM
+
+```html
+<div class="fui-dialog-header">
+  <span>{title}</span>
+  <button aria-label="close" />
+</div>
+```
+
+### API
 
-_Provide some representative example code that uses the proposed API for the component_
+| Property          | Values  | Default     | Purpose                         |
+| ----------------- | ------- | ----------- | ------------------------------- |
+| `closeButton`     | _slot_  | `undefined` | Button that closes the dialog   |
+| `hideCloseButton` | boolean | `false`     | Shows or hides the close button |
 
-## Variants
+## Dialog Body
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+The body is a container where the content of the dialog is rendered. Apart from padding, this component does not have other behaviour.
 
-## API
+```html
+<div class="fui-dialog-body">{children}</div>
+```
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+## Dialog Footer
 
-## Structure
+The footer is a container for the actions of the dialog, which must be not more than 3 (primary, secondary and tertiary actions).
 
-- _**Public**_
-- _**Internal**_
-- _**DOM** - how the component will be rendered as HTML elements_
+### DOM
+
+```html
+<div class="fui-dialog-footer">
+  <button>{tertiaryAction}</button>
+  <button>{primaryAction}</button>
+  <button>{secondaryAction}</button>
+</div>
+```
+
```

Comment:
Missing API table for DialogFooter

```suggestion

###API

| Property          | Values  | Default     | Purpose                         |
| ----------------- | ------- | ----------- | ------------------------------- |
| `primaryAction`     | _slot_  | `undefined` | |
| `secondaryAction`     | _slot_  | `undefined` | |
| `tertiaryAction`     | _slot_  | `undefined` | |

```

Reply:
not exactly! I did not include this on purpose as it is something still in discussion with design. The issue is this:
In Office the order of the action buttons in footer are 1. primary button (on the left) and 2. secondary button (on the right)
![OFFICE](https://user-images.githubusercontent.com/12565826/158567930-24efaf61-4d31-4777-baea-48cecea87b4b.png)
which would mean this and it is correct
```
<DialogFooter primaryAction={send()} secondaryAction={cancel()} />
```
BUT, in Teams, there is the oposite:
![teams](https://user-images.githubusercontent.com/12565826/158568272-f8018941-df8a-46eb-b850-816d7aef21a3.png)
and that would mean:
which would mean this:
```
<DialogFooter primaryAction={cancel()} secondaryAction={send()} />
```
which is not quite right, right?  As it is weird which should be primary action and which the secondary action. This also affects the tab order when navigating by keyboard. 
This is still an open discussion with the design team and waiting more information on this. But if you have any ideas how to address this, I would highly appreciate them ^_^

Reply:
That makes sense, thanks for clarifying! From my point of view, we could always just provide the positioning slots and let the users decide which type of Button/button variant they want to apply.